bigtable/tests/mutations_integration_test.cc

```diff
@@ -0,0 +1,555 @@
+                           cell.labels());
+    expected_cells_ignore_time.emplace_back(std::move(newCell));
+  }
+  for (auto& cell : actual_cells) {
+    bigtable::Cell newCell(cell.row_key(), cell.family_name(),
+                           cell.column_qualifier(), 1000, cell.value(),
+                           cell.labels());
+    actual_cells_ignore_time.emplace_back(std::move(newCell));
+  }
+
+  CheckEqualUnordered(expected_cells_ignore_time, actual_cells_ignore_time);
+}
+
+/**
+ * Verify that the deletion of records for specific row_key, column_family,
+ * column_identifier and within the time range are deleted from Cloud
+ * Bigtable.
+ */
+TEST_F(MutationIntegrationTest, DeleteFromColumnForTimestampRangeTest) {
+  std::string const table_name = "table-delete-for-column-time-range";
+
+  auto table = CreateTable(table_name, table_config);
+
+  // Create a vector of cell which will be inserted into bigtable
+  std::string const row_key = "DelColKey";
+  std::vector<bigtable::Cell> created_cells{
+      {row_key, column_family1, "column_id1", 0, "v-c-0-0", {}},
+      {row_key, column_family1, "column_id2", 1000, "v-c-0-1", {}},
+      {row_key, column_family1, "column_id3", 2000, "v-c-0-2", {}},
+      {row_key, column_family2, "column_id2", 2000, "v-c0-0-0", {}},
+      {row_key, column_family2, "column_id2", 1000, "v-c0-0-1", {}},
+      {row_key, column_family2, "column_id2", 3000, "v-c0-0-2", {}},
+      {row_key, column_family2, "column_id2", 4000, "v-c0-0-3", {}},
+      {row_key, column_family2, "column_id3", 1000, "v-c1-0-1", {}},
+      {row_key, column_family2, "column_id2", 2000, "v-c1-0-2", {}},
+      {row_key, column_family3, "column_id1", 2000, "v-c1-0-2", {}},
+  };
+
+  // timestamp_end is not inclusive.
+  std::int64_t timestamp_begin = 2000;
+  std::int64_t timestamp_end = 4000;
+  std::vector<bigtable::Cell> expected_cells{
+      {row_key, column_family1, "column_id1", 0, "v-c-0-0", {}},
+      {row_key, column_family1, "column_id2", 1000, "v-c-0-1", {}},
+      {row_key, column_family1, "column_id3", 2000, "v-c-0-2", {}},
+      {row_key, column_family2, "column_id2", 1000, "v-c0-0-1", {}},
+      {row_key, column_family2, "column_id2", 4000, "v-c0-0-3", {}},
+      {row_key, column_family2, "column_id3", 1000, "v-c1-0-1", {}},
+      {row_key, column_family3, "column_id1", 2000, "v-c1-0-2", {}},
+  };
+
+  // Create records
+  CreateCells(*table, created_cells);
+
+  // Delete the columns with column identifier as column_id2
+  DeleteFromColumn(*table, row_key, column_family2, "column_id2",
+                   timestamp_begin, timestamp_end);
+
+  auto actual_cells = ReadRows(*table, bigtable::Filter::PassAllFilter());
+
+  DeleteTable(table_name);
+
+  CheckEqualUnordered(expected_cells, actual_cells);
+}
+
+/**
+ * Verify that DeleteFromColumn operation for invalid time range
+ * should throw a range exception
+ */
+TEST_F(MutationIntegrationTest, DeleteFromColumnForTimestampRangeValidityTest) {
```

Comment:
It seems to me that this only validates that `DeleteFromColumn()` raises, locally, this add no more testing than what the unit tests did.  Why not remove it?

Reply:
"DeleteFromColumnForTimestampRangeValidityTest": I thought to write this test case by considering that if timestamp is in wrong range. i.e. time_begin > time_end, then this function raises the exception of "range_error". 
whereas
"DeleteFromColumnForTimestampRangeTest": This test case deletes records which are in the range. So these two test cases are handling different criteria.
"DeleteFromColumnForAllTest": This test case deletes all records for the specific column identifier without considering the timestamp.
So I think these three test cases are handling different criteria.
What do you think about this. Shall I remove the test case of checking throw exception.

Reply:
The problem is that this test calls the `DeleteFromColumn()` function in this file.  The [first thing](https://github.com/GoogleCloudPlatform/google-cloud-cpp/pull/282/files/29bc245c6fc9de7f4afa12bc8318fc2d8de646c2#diff-3891e80405b38b61169f246074d884d6R110) this function does is call `bigtable::DeleteFromColumn()`.  That function will raise, and nothing else in the function will execute.

But we already have a unit test to verify that `bigtable::DeleteFromColumn()` raises.

Please remove the `DeleteFromColumnForTimestampRangeValidityTest` test case.